scripts/apps/desks/controllers/DeskConfigController.ts

```diff
@@ -61,4 +73,70 @@ export function DeskConfigController($scope, $controller, notify, desks, WizardH
     $scope.getDeskUsers = function(desk) {
         return desks.deskMembers[desk._id];
     };
+
+    const closeModel = function() {
+        $scope.modalActive = false;
+        $scope.step.current = null;
+        $scope.desk.edit = null;
+    };
+
+    const calculateDiff = function() {
+        let diff = _.extend({}, $scope.desk.edit);
+
+        if (angular.isDefined($scope.desk.orig)) {
+            angular.forEach(_.keys(diff), (key) => {
+                if (_.isEqual(diff[key], $scope.desk.orig[key])
+                    || (key === 'content_expiry' && $scope.desk.orig[key] === null && diff[key] === 0)) {
+                    delete diff[key];
+                }
+            });
+        }
+        return diff;
+    };
+
+    $scope.confirmSave = function(diff, step, close?) {
+        modal.confirm(
+            gettext('You have unsaved changes in {{tab}} tab. Do you want to save them now?', {tab: step}),
+            gettext('Save changes?'),
+            gettext('Save'),
+            gettext('Ignore'))
+            .then(() => {
+                desks.save($scope.desk.orig, diff).then((res) => {
+                    _.merge($scope.desk.edit, res);
+                    _.merge($scope.desk.orig, res);
+                    if (diff.members) {
```

Comment:
Is it possible to avoid using the field names when updating the object? How about iterating all members and updating them?

Reply:
Yyou mean to update the members every time even if they were not changed?

Reply:
When I look at this again, I think it would probably be best to avoid dealing with members explicitly at all. Just apply the entire diff and that's it. If members haven't changed, they shouldn't be in a diff.